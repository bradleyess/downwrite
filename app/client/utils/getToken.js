import Cookies from 'universal-cookie'

export default (req, query) => {
	const ck = new Cookies()

	const token = req ? req.universalCookies.cookies.DW_TOKEN : query.token || ck.get('DW_TOKEN')
	const userID = req ? req.universalCookies.cookies.DW_USERID : ck.get('DW_USERID')
	const username = req ? req.universalCookies.cookies.DW_USERNAME : ck.get('DW_USERNAME')

	return {
		token,
		userID,
		username
	}
}
