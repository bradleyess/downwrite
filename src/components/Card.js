import React from 'react'
import { Flex, Block } from 'glamor/jsxstyle'
import { Link } from 'react-router-dom'
import { css } from 'glamor'
import distance from 'date-fns/distance_in_words_to_now'

const s = {
	title: css({
		fontSize: 14,
		marginBottom: 0,
		fontWeight: 700,
		'@media (min-width: 57.75rem)': { fontSize: 18 }
	}),
	delete: css({
		color: 'var(--color-2)',
		border: 0,
		background: 'none',
		appearance: 'none',
		WebkitFontSmoothing: 'antialiased'
	}),
	content: css({
		fontSize: 12
	}),
	card: css({
		fontWeight: 400,
		boxShadow: '0 0 2px rgba(0,0,0,.07), 0 2px 4px rgba(0,0,0,.12)',
		backgroundColor: 'white'
	}),
	meta: css({
		fontSize: 12,
		display: 'block',
		color: '#757575',
		fontWeight: 400
	}),
	tray: css({
		'& a': {
			color: '#4382A1'
		}
	})
}

const Card = ({ title, id, content, dateAdded, onDelete }) => (
	<Flex
		height={192}
		flexDirection="column"
		justifyContent="space-between"
		className={css(s.card)}
		data-test="Card">
		<Block borderBottom="1px solid #DBDCDD" padding="12px 8px">
			<h2 className={css(s.title)}>
				<Link to={`/${id}/edit`}>{title}</Link>
			</h2>
			<small className={css(s.meta)}>added {distance(dateAdded)} ago</small>
		</Block>

		<Block padding={8} flex={1} overflow="hidden">
			{content && (
				<p data-test="snippet" className={css(s.content)}>
					{content.blocks[0].text.substr(0, 75)}
				</p>
			)}
		</Block>
		<Flex
			className={css(s.tray)}
			padding={8}
			fontWeight={700}
			fontSize={12}
			justifyContent="space-between"
			backgroundColor="rgba(101, 163, 191, .125)">
			<Link to={`/${id}/edit`}>Edit</Link>

			{onDelete && (
				<button className={css(s.delete)} onClick={onDelete}>
					Delete
				</button>
			)}
		</Flex>
	</Flex>
)

export default Card
