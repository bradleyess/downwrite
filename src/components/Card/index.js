import React from 'react'
import { Flex, Block } from 'glamor-jsxstyle'
import { Link } from 'react-router-dom'
import { css } from 'glamor'

const s = {
  title: css({
    fontSize: 24,
    fontWeight: 600
  }),
  body: css({
    fontSize: `small`,
    opacity: 0.875,
    padding: 16
  }),
  card: css({
    borderLeftWidth: 8,
    borderLeftStyle: 'solid',
    borderColor: `var(--color-1)`,
    boxShadow: '0 0 2px rgba(0,0,0,.07), 0 2px 4px rgba(0,0,0,.12)',
    backgroundColor: 'white'
  }),
  meta: css({
    opacity: .5
  })
}

const Card = ({ title, id, content, author, className, ...props }) => (
  <Block component='li' padding='0 20px' className={className}>
    <Block className={css(s.card)} {...props}>
      <Flex borderBottom='1px solid #f2f2f2' padding={16} justifyContent='space-between' alignItems='flex-start'>
        <div>
          <h2 className={s.title}>{title}</h2>
          <small className={s.meta}>
            {author} - {id}
          </small>
        </div>
        <Link to={`edit/${id}`}>Edit</Link>
      </Flex>
      {content && <p className={s.body}>{content.slice(0, 90)}...</p>}
    </Block>
  </Block>
)

export default Card
