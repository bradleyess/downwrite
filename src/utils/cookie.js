// @flow
import Cookies from 'universal-cookie'
const oatmeal: typeof Cookies = new Cookies()

export const TOKEN: string = oatmeal.get('DW_TOKEN')
export const USER_ID: string = oatmeal.get('DW_USERID')
export const USER_NAME: string = oatmeal.get('DW_USERNAME')

export const signOut: Function = () => {
	oatmeal.remove('DW_TOKEN')
	oatmeal.remove('DW_USERID')
	oatmeal.remove('DW_USERNAME')

}

export const signIn: Function = (token, id, name) => {
	oatmeal.set('DW_TOKEN', token)
	oatmeal.set('DW_USERID', id)
	oatmeal.set('DW_USERNAME', name)
}

export default oatmeal

export const intialState = {
	token: TOKEN,
	user: USER_ID
}
