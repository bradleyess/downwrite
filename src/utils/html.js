import { Html } from 'slate'

const BLOCK_TAGS = {
  blockquote: 'quote',
  p: 'paragraph',
  pre: 'code'
}

// Add a dictionary of mark tags.
const MARK_TAGS = {
  em: 'italic',
  strong: 'bold',
  u: 'underline',
}

const rules = [
  {
    deserialize(el, next) {
      const type = BLOCK_TAGS[el.tagName]
      if (!type) return
      return {
        kind: 'block',
        type: type,
        nodes: next(el.children)
      }
    },
    serialize(object, children) {
      if (object.kind != 'block') return
      switch (object.type) {
        case 'code': return <pre><code>{children}</code></pre>
        case 'paragraph': return <p>{children}</p>
        case 'quote': return <blockquote>{children}</blockquote>
      }
    }
  },
  // Add a new rule that handles marks...
  {
    deserialize(el, next) {
      const type = MARK_TAGS[el.tagName]
      if (!type) return
      return {
        kind: 'mark',
        type: type,
        nodes: next(el.children)
      }
    },
    serialize(object, children) {
      if (object.kind != 'mark') return
      switch (object.type) {
        case 'bold': return <strong>{children}</strong>
        case 'italic': return <em>{children}</em>
        case 'underline': return <u>{children}</u>
      }
    }
  }
]

const html = new Html({ rules })

export default html
