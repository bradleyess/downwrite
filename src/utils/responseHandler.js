import { convertFromRaw } from 'draft-js'

type Handler = {
	json: Function,
	statusText: string,
	ok: boolean
}

export const handleResponse: Function = (response: Handler) => {
	if (response.ok) {
		return response.json()
	} else {
		let error: typeof Error = new Error(response.statusText)
		error.response = response
		throw error
	}
}

type Content = { blocks: array, entityMap: Object }

export const superConverter: Function = (content: Content) => {
	return content.hasOwnProperty('entityMap')
		? convertFromRaw(content)
		: convertFromRaw({ blocks: content.blocks, entityMap: {} })
}
