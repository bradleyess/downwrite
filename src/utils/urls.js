//@flow

export const POST_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:4411/posts'
		: 'https://api.downwrite.us/posts'


export const PREVIEW_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:4411/posts/preview'
		: 'https://api.downwrite.us/posts/preview'

export const USER_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:4411/users'
		: 'https://api.downwrite.us/users'

export const AUTH_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:4411/users/authenticate'
		: 'https://api.downwrite.us/users/authenticate'

export const MD_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:8793/'
		: 'https://markdown.downwrite.us/'

export const JSON_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:8794/'
		: 'https://json.downwrite.us/'

export const MEDIUM_ENDPOINT: string =
	process.env.NODE_ENV === 'development'
		? 'http://localhost:6335/'
		: 'https://medium.downwrite.us/'
