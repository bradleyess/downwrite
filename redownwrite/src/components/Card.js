import React from 'react'
import styled from 'styled-components'
import Link from 'react-router-dom/Link'
import distance from 'date-fns/distance_in_words_to_now'
import { colors } from '../utils/defaultStyles'

const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 192px;
  font-weight: 400;
  box-shadow: 0 0 2px rgba(0, 0, 0, 0.07), 0 2px 4px rgba(0, 0, 0, 0.12);
  background-color: white;
`

const CardTitle = styled.h2`
  font-size: 14px;
  margin-bottom: 0px;
  font-weight: 700;

  @media (min-width: 57.75rem) {
    font-size: 18px;
  }
`

const CardHeader = styled.header`
  display: block;
  border-bottom: 1px solid ${colors.gray200};
  padding: 12px 8px;
`

const CardMeta = styled.small`
  font-size: 12px;
  display: block;
  color: ${colors.gray300};
  font-weight: 400;
`

const CardTray = styled.footer`
  display: flex;
  padding: 8px;
  font-weight: 700;
  font-size: 12px;
  justify-content: space-between;
  background-color: rgba(101, 163, 191, 0.125);

  & a:hover {
    color: ${colors.blue500};
  }
`

const CardPreview = styled.p`
  font-size: 12px;
`

const CardBody = styled.div`
  display: block;
  padding: 0.5rem;
  flex: 1;
  overflow: hidden;
`

const CardDelete = styled.button`
  color: ${colors.blue700};
  border: 0px;
  background: none;
  appearance: none;
  -webkit-font-smoothing: antialiased;
`

const SxLink = styled(Link)`
  margin-right: 8px;
`

const Card = ({ title, id, content, dateAdded, onDelete, ...args }) => (
  <CardContainer data-testid="CARD">
    <CardHeader>
      <CardTitle data-testid="CARD_TITLE">
        <Link to={`/${id}/edit`}>{title}</Link>
      </CardTitle>
      <CardMeta>added {distance(dateAdded)} ago</CardMeta>
    </CardHeader>

    <CardBody>
      {content && (
        <CardPreview data-test="snippet">{content.blocks[0].text.substr(0, 75)}</CardPreview>
      )}
    </CardBody>
    <CardTray>
      <div data-testid="CARD_EXCERPT">
        <SxLink to={`/${id}/edit`}>Edit</SxLink>
        {args.public && <Link to={`/${id}/preview`}>Preview</Link>}
      </div>
      {onDelete && (
        <CardDelete data-testid="CARD_DELETE_BUTTON" onClick={onDelete}>
          Delete
        </CardDelete>
      )}
    </CardTray>
  </CardContainer>
)

export default Card
