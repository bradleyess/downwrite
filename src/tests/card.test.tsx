import * as React from "react";
import { fireEvent, render, waitForElement } from "@testing-library/react";
import Card from "../components/card";
import { createMockPost } from "../utils/testing";
import { MemoryRouter } from "react-router-dom";

const title = "Starting Again";
const post = createMockPost({ title, id: "4444" });
const mockDelete = jest.fn();

describe("<Card />", () => {
  it("contains snippet from content", () => {
    const TestCard = render(
      <MemoryRouter>
        <Card public={false} {...post} onDelete={mockDelete} />
      </MemoryRouter>
    );
    const snippet = TestCard.getByTestId("CARD_EXCERPT").textContent;
    expect(snippet.length).toBeLessThanOrEqual(90);
  });

  it("contains a title", () => {
    const TestCard = render(
      <MemoryRouter>
        <Card public={false} {...post} onDelete={mockDelete} />{" "}
      </MemoryRouter>
    );
    expect(post.title).toBe(title);
    expect(TestCard.getByTestId("CARD_TITLE")).toHaveTextContent(title);
  });

  it("contains delete button", async () => {
    const TestCard = render(
      <MemoryRouter>
        <Card public={false} {...post} onDelete={mockDelete} />
      </MemoryRouter>
    );
    const deleteButton = await waitForElement(() =>
      TestCard.getByTestId("CARD_DELETE_BUTTON")
    );
    expect(deleteButton).toBeInTheDocument();
    fireEvent.click(deleteButton);

    expect(mockDelete).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const TestCard = render(
      <MemoryRouter>
        <Card public={false} {...post} onDelete={mockDelete} />
      </MemoryRouter>
    );
    expect(TestCard.container.firstChild).toMatchSnapshot();
  });
});
