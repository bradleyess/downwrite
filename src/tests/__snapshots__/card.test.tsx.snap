// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
<div
  class="Sheet Card"
  data-testid="CARD"
>
  <header
    class="CardHeader"
  >
    <h2
      class="CardTitle"
      data-testid="CARD_TITLE"
    >
      <a
        href="/edit/4444"
      >
        Starting Again
      </a>
    </h2>
    <small
      class="CardMeta"
    >
      added 
      10 months
       ago
    </small>
  </header>
  <footer
    class="CardTray"
  >
    <div
      class="links"
      data-testid="CARD_EXCERPT"
    >
      <a
        href="/edit/4444"
        style="margin-right: 8px;"
      >
        Edit
      </a>
    </div>
    <button
      class="CardDeleteButton"
      data-testid="CARD_DELETE_BUTTON"
    >
      Delete
    </button>
  </footer>
</div>
`;
